attachment_preview/static/src/js/models/attachment_card/attachment_card.esm.js

```diff
@@ -0,0 +1,276 @@
+                                    id,
+                                    attachments[id].url,
+                                    extensions[id],
+                                    attachments[id].title
+                                ),
+                            };
+                        }
+                    );
+                    chatterpreviewableAttachments = reviewableAttachments;
+                });
+            },
+
+            /**
+             * @private
+             * @param {attachment_id} attachment_id of the attachment
+             * @param {attachment_url} attachment_url of the attachment
+             * @param {attachment_extension} attachment_extension of the attachment
+             * @param {attachment_title} attachment_title of the attachment
+             * @param {split_screen} split_screen of the attachment
+             */
+            _showPreview(
+                attachment_id,
+                attachment_url,
+                attachment_extension,
+                attachment_title,
+                split_screen
+            ) {
+                // Let active_attURL = "";
+                // this.attachmentList.attachments.forEach((att) => {
+                //     if (
+                //         parseInt(att.localId.slice(20).slice(0, -1)) === attachment_id
+                //     ) {
+                //         if (att.__values.url === undefined) {
+                //             att.__values.url = attachment_url.slice(
+                //                 window.location.origin.length
+                //             );
+                //             active_attURL = att.__values.url;
+                //         }
+                //     }
+                // });
+                var url = getUrl(
+                    attachment_id,
+                    attachment_url,
+                    attachment_extension,
+                    attachment_title
+                );
+
+                if (split_screen) {
+                    this.component.trigger("onAttachmentPreview", {
+                        url: url,
+                        active_attachment_id: active_attachment_id,
+                    });
+                } else {
+                    window.open(url);
+                }
+            },
+
+            /**
+             * @private
+             * @param {event} event
+             */
+            _onPreviewAttachment(event) {
+                event.preventDefault();
+
+                var self = this,
+                    $target = $(event.currentTarget),
+                    split_screen = $target.attr("data-target") !== "new",
+                    attachment_id = this.attachment.id,
+                    attachment_extension = "pdf",
+                    attachment_title = this.attachment.filename,
+                    attachment_url = this.attachment.defaultSource;
+                active_attachment_id = attachment_id;
+
+                if (attachment_extension) {
```

Comment:
The if else path Lines 250 to 272 can be replaced by:
```
                rpc.query({
                    model: "ir.attachment",
                    method: "get_attachment_extension",
                    args: [attachment_id],
                }).then(function (extension) {
                    self._showPreview(
                        attachment_id,
                        attachment_url,
                        attachment_extension,
                        extension,
                        attachment_title,
                        split_screen
                    );
                });
```

Reply:
it's done!